aiohomekit/controller/ble/pairing.py

```diff
@@ -467,8 +476,10 @@ async def _process_disconnected_events(self) -> None:
                 self.rssi,
             )
             try:
-                protocol_param = await self.get_all_protocol_params()
-                results = await self.get_characteristics(list(self.subscriptions))
+                (
+                    protocol_param,
+                    results,
+                ) = await self._process_disconnected_events_with_retry()
```

Comment:
I split this out so it was only wrapped once instead of two functions with their own wrappers so we don't retry twice, and then if it does raise we are still handling the exception